ledger-core/virtual/handlers/vfindcallrequest.go

```diff
@@ -0,0 +1,191 @@
+}
+
+/* -------- Declaration ------------- */
+
+var dSMVFindCallRequestInstance smachine.StateMachineDeclaration = &dSMVFindCallRequest{}
+
+type dSMVFindCallRequest struct {
+	smachine.StateMachineDeclTemplate
+}
+
+func (*dSMVFindCallRequest) InjectDependencies(sm smachine.StateMachine, _ smachine.SlotLink, injector *injector.DependencyInjector) {
+	s := sm.(*SMVFindCallRequest)
+
+	injector.MustInject(&s.pulseSlot)
+	injector.MustInject(&s.messageSender)
+}
+
+func (*dSMVFindCallRequest) GetInitStateFor(sm smachine.StateMachine) smachine.InitFunc {
+	s := sm.(*SMVFindCallRequest)
+	return s.Init
+}
+
+/* -------- Instance ------------- */
+
+func (s *SMVFindCallRequest) GetStateMachineDeclaration() smachine.StateMachineDeclaration {
+	return dSMVFindCallRequestInstance
+}
+
+func (s *SMVFindCallRequest) Init(ctx smachine.InitializationContext) smachine.StateUpdate {
+	if s.pulseSlot.State() == conveyor.Present {
+		ctx.SetDefaultMigration(s.migrateFutureMessage)
+		return ctx.Jump(s.stepWait)
+	}
+
+	return ctx.Jump(s.stepProcessRequest)
+}
+
+func (s *SMVFindCallRequest) stepWait(ctx smachine.ExecutionContext) smachine.StateUpdate {
+	return ctx.Sleep().ThenRepeat()
+}
+
+func (s *SMVFindCallRequest) migrateFutureMessage(ctx smachine.MigrationContext) smachine.StateUpdate {
+	ctx.SetDefaultMigration(func(ctx smachine.MigrationContext) smachine.StateUpdate {
+		return ctx.Stop()
+	})
+	return ctx.Jump(s.stepProcessRequest)
+}
+
+func (s *SMVFindCallRequest) stepProcessRequest(ctx smachine.ExecutionContext) smachine.StateUpdate {
+	accessor, isCondPublished := callsummary.GetSummarySMSyncAccessor(ctx, s.Payload.Callee)
+	if isCondPublished {
+		// if cond is still published we need to wait for object to register itself in call summary
+		s.finishedAccessor = accessor
+		return ctx.Jump(s.stepWaitCallResult)
+	}
+	return ctx.Jump(s.stepGetRequestData)
+}
+
+func (s *SMVFindCallRequest) stepWaitCallResult(ctx smachine.ExecutionContext) smachine.StateUpdate {
+	var cond smachine.SyncLink
+
+	switch s.finishedAccessor.Prepare(func(link *smachine.SyncLink) {
+		cond = *link
+	}).TryUse(ctx).GetDecision() {
+	case smachine.NotPassed:
+		return ctx.WaitShared(s.finishedAccessor.SharedDataLink).ThenRepeat()
+	case smachine.Impossible:
+		panic(throw.NotImplemented())
+	case smachine.Passed:
+		// go further
+	default:
+		panic(throw.Impossible())
+	}
+
+	if ctx.Acquire(cond).IsNotPassed() {
+		return ctx.Sleep().ThenRepeat()
+	}
+
+	return ctx.Jump(s.stepGetRequestData)
+}
+
+func (s *SMVFindCallRequest) stepGetRequestData(ctx smachine.ExecutionContext) smachine.StateUpdate {
+	summarySharedStateAccessor, ok := callsummary.GetSummarySMSharedAccessor(ctx, s.pulseSlot.PulseData().PulseNumber)
+	// no result
+	if !ok {
+		s.status = payload.MissingCall
+		if s.Payload.Outgoing.GetLocal().Pulse() < s.pulseSlot.PulseData().PulseNumber {
+			s.status = payload.UnknownCall
+		}
+
+		return ctx.Jump(s.stepSendResponse)
+	}
+
+	var (
+		notFound bool
+	)
```

Comment:
`var notFound `
more simple?

Reply:
One more point, `found=true/false` more clear for me, 'notFound=false' is reversed `found=true`

Reply:
done